14121 update plugin development docs for pyproject.toml (#15952)
netbox-community/netbox: docs/plugins/development/index.md

## Create setup.py

> `pyproject.toml` is the [configuration file](https://packaging.python.org/en/latest/guides/writing-pyproject-toml/) used to package and install our plugin once it's finished. The primary function of this script is to call the build system to create a Python distribution package. We can pass a number of keyword arguments to control the package creation as well as to provide metadata about the plugin. An example `pyproject.toml` is below:

Reviewer: > The primary function of this script

Maybe a bid pedantic, but this is no longer a script. We should more accurately explain how the `.toml` file is processed.

```
# See PEP 518 for the spec of this file
# https://www.python.org/dev/peps/pep-0518/

[build-system]
requires = ["setuptools"]
build-backend = "setuptools.build_meta"

[project]
name =  "my-example-plugin"
version = "0.1.4"
authors = [

>     {name = "Jeremy Stretch", email = "test@netboxlabs.com"},

Reviewer: Let's take this opportunity to use someone else's name here. :laughing:

]
description = "An example NetBox plugin."
readme = "README.md"

classifiers=[
    'Development Status :: 3 - Alpha',
    'Intended Audience :: Developers',
    'Natural Language :: English',
    "Programming Language :: Python :: 3 :: Only",

>     'Programming Language :: Python :: 3.9',

Reviewer: Might as well drop Python 3.9 from the example since NetBox v4.0 requires 3.10+.

> ## Create setup.py

Reviewer: Looks like there are still some latent references to `setup.py` here, above in the discussion of the file structure, and below in the development installation section.

`pyproject.toml` is the [configuration file](https://packaging.python.org/en/latest/guides/writing-pyproject-toml/) used to package and install our plugin once it's finished. It is used by packaging tools, as well as other tools. The primary function of this file is to call the build system to create a Python distribution package. We can pass a number of keyword arguments to control the package creation as well as to provide metadata about the plugin. There are three possible TOML tables in this file:

> * The [build-system] Allows you to declare which build backend you use and which other dependencies are needed to build your project.
> * The [project] table is the format that most build backends use to specify your project’s basic metadata, such as the dependencies, your name, etc.
> * The [tool] table has tool-specific subtables, e.g., [tool.black], [tool.mypy]. Consult the particular tool’s documentation to know what it can contain.

Reviewer: ```suggestion
* `[build-system]` allows you to declare which build backend you use and which other dependencies (if any) are needed to build your project.
* `[project]` is the format that most build backends use to specify your project’s basic metadata, such as the author's name, project URL, etc.
* `[tool]` has tool-specific subtables, e.g., `[tool.black]`, `[tool.mypy]`. Consult the particular tool’s documentation for reference.
```
